app/src/test/java/com/codesoom/assignment/product/domain/ProductTest.java

```diff
@@ -0,0 +1,122 @@
+package com.codesoom.assignment.product.domain;
+
+import com.codesoom.assignment.ProvideInvalidProductArguments;
+import com.codesoom.assignment.product.exception.ProductInvalidFieldException;
+import com.codesoom.assignment.product.exception.ProductInvalidPriceException;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.List;
+
+import static com.codesoom.assignment.Constant.IMAGE_URL;
+import static com.codesoom.assignment.Constant.MAKER;
+import static com.codesoom.assignment.Constant.NAME;
+import static com.codesoom.assignment.Constant.OTHER_IMAGE_URL;
+import static com.codesoom.assignment.Constant.OTHER_MAKER;
+import static com.codesoom.assignment.Constant.OTHER_NAME;
+import static com.codesoom.assignment.Constant.OTHER_PRICE;
+import static com.codesoom.assignment.Constant.PRICE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DisplayName("CatToy 객체 유닛 테스트")
+class ProductTest {
+
+    @DisplayName("객체를 생성할 수 있습니다.")
+    @Test
+    void create() {
+        final Product product = Product.of(NAME, MAKER, PRICE, IMAGE_URL);
+
+        assertThat(product.getName()).isEqualTo(NAME);
+        assertThat(product.getPrice()).isEqualTo(PRICE);
+        assertThat(product.getMaker()).isEqualTo(MAKER);
+        assertThat(product.getImageUrl()).isEqualTo(IMAGE_URL);
+    }
+
+    @DisplayName("유효하지 않은 가격으로 객체를 만들 수 없습니다.")
+    @Test
+    void createWithInvalidPrice() {
+
+        assertThatThrownBy(() -> Product.of(NAME, MAKER, -5000L, IMAGE_URL)
+        ).isInstanceOf(ProductInvalidPriceException.class);
+    }
```

Comment:
이 테스트는 유효하지 않은 가격의 예로 `-5000L` 하나만을 제안하고 있어요. `-1`이나 `-2000000`도 유효하지 않은 가격이라는 것은 추측을 통해서만 알 수 있습니다. 물론 귀찮게 `Product` 클래스를 열어 코드를 읽으면 알 수 있겠죠. 하지만 코드를 읽는 사람이 그렇게 행동하도록 유도하는 코드는 좋은 코드에서 거리가 있겠죠?

추측을 더 쉽게 만들어 주기 위해 풍부한 예제를 제공해 봅시다. `List`를 쓸 수도 있겠죠.

```suggestion
    @DisplayName("유효하지 않은 가격으로 객체를 만들 수 없습니다.")
    @Test
    void createWithInvalidPrice() {
        List<Long> invalidPrice = List.of(0L, -1L, -5000L, Long.MIN_VALUE);

        for(final Long price : invalidPrice) {
            assertThatThrownBy(() -> Product.of(NAME, MAKER, price, IMAGE_URL)
            ).isInstanceOf(ProductInvalidPriceException.class);
        }

    }
```

테스트 코드에서 한 가지 값으로만 테스트하는 것은 좋지 않습니다. 테스트 가능한 값은 항상 단일 값이 아니라 특정 범위에 있을 수 있다는 것을 기억해 두세요.